delivery_schenker/models/delivery_carrier.py

```diff
@@ -241,20 +264,19 @@ def _prepare_schenker_address(
             "street": partner.street,
             "postalCode": partner.zip,
             "city": partner.city,
-            "stateCode": partner.state_id.code,
-            "stateName": partner.state_id.name,
             "countryCode": partner.country_id.code,
             "preferredLanguage": self.env["res.lang"]._lang_get(partner.lang).iso_code,
         }
         # Optional stuff. The API doesn't like falsy or empty request fields
-        if partner.email:
-            vals["email"] = partner.email
-        if partner.mobile:
-            vals["mobilePhone"] = partner.mobile
-        if partner.phone:
-            vals["phone"] = partner.phone
-        if partner.street2:
-            vals["street2"] = partner.street2
+        for schenker_key, expression in self._schenker_address_optional_fields():
+            fields = expression.split(".")
+            value = getattr(partner, fields[0])
```

Comment:
Why doing it in 2 steps? It can be in the same loop.

Reply:
Sorry i don't know what you mean by that

Reply:
ping @pedrobaeza :-)

Reply:
You are repeating operations for index 0 and then a loop for [1:]. Do everything in the same loop for code readability and for saving lines.